### ISSUE TITLE
Request support for remote IPMI

### ISSUE THREAD

**Issue by davidnghongkong** [petersulyok/smfc#27]
it seems like this script is install on a host directly.
but what if the host is proxmox and want to keep it clean, so this script should install in a vm.
possible add remote support?
I think the command is
`ipmitool -U ipmi_user_name -P ipmi_password -H ipmi_ip`

**Comment by petersulyok** [comment]
Not sure if I got the idea correctly but I think it could not work like this.
Remote IPMI management is possible (as you suggested) but you need CPU and HD temperature as well from the remote machine and it is not trivial.

I would rather install this script on `proxmox` (in `/opt` or in `/usr/local` folder). I do not see other ways.

**Comment by petersulyok** [comment]
Hi @davidngrc, thinking further on this feature request I think adding extra command line parameter would be not hard at all. Could you please double check if you can read the temperature of the CPU and SATA HD correctly in a VM? I'm not sure if `drivetemp` can be loaded properly and please compare the temperature value coming in a VM with the temperature of the host machine. Thanks.

**Comment by petersulyok** [comment]
Adding a new docker support for SMFC v3.4.0 might solve your issue here.
@davidngrc please check it out.

**Comment by alexanderek** [comment]
Hello!

In my case, the script should work on Proxmox host, getting CPU temperature locally, but HDD disks themselves are in TrueNAS VM. Is there any way to make the script work?

**Comment by petersulyok** [comment]
> In my case, the script should work on Proxmox host, getting CPU temperature locally, but HDD disks themselves are in TrueNAS VM. Is there any way to make the script work?

Let me understand your plan. You are running Proxmox host and there is a VM running TrueNAS with assigned hard disks, right? My assumptions is that you may run `smfc` inside TrueNAS. What do you think?

**Comment by petersulyok** [comment]
Also, my other idea is [to run docker in LCX](https://www.google.com/search?q=proxmox+how+to+run+docker&sca_esv=d3441e0a04f98502&ei=Un3UZZLwBrjs7_UPwsi3mA8&udm=&ved=0ahUKEwiS4fip2bmEAxU49rsIHULkDfMQ4dUDCBA&uact=5&oq=proxmox+how+to+run+docker&gs_lp=Egxnd3Mtd2l6LXNlcnAiGXByb3htb3ggaG93IHRvIHJ1biBkb2NrZXIyBhAAGAgYHkj9CVCLBljeB3ABeAGQAQCYAXygAbMBqgEDMS4xuAEDyAEA-AEBwgIKEAAYRxjWBBiwA4gGAZAGCA&sclient=gws-wiz-serp) in Proxmox.

**Comment by alexanderek** [comment]
Correct, proxmox and vm truenas scale. I originally thought of running this on the proxmox host, and getting disk temperatures somehow remotely from truenas.

**Comment by petersulyok** [comment]
I do not have experience of running `smfc` on either Proxmox or on TrueNAS Scale platforms. 
I thought docker support might help here, but if you don't access to the host hardware and IPMI interface this could be a issue, definitely. 

I will try to understand the issues around these two platforms better.

**Comment by duecedriver** [comment]
> Hi [@davidngrc](https://github.com/davidngrc), thinking further on this feature request I think adding extra command line parameter would be not hard at all. Could you please double check if you can read the temperature of the CPU and SATA HD correctly in a VM? I'm not sure if `drivetemp` can be loaded properly and please compare the temperature value coming in a VM with the temperature of the host machine. Thanks.

this would be easy.. 

running ipmi remote commands all you would need would be to query during setup the account name and password and address of the impi interface.  every ipmi command could be send with these values without logging into impi in a session

an additional benefit ..

you could have a single container/host control multiple servers

**Comment by bockbilbo** [comment]
I am in the same situation as @alexanderek. Passing my SATA controller to a TrueNAS Scale VM in Proxmox on a host with an X11SSL-F mother board. I'd like to figure out how to make smfc work in such scenario, as the docker container within LXC in Proxmox won't have access to the drives. 

@alexanderek did you figure this out? @petersulyok do you need support understanding the platform or testing?

Thanks!

**Comment by petersulyok** [comment]
I will look into this how can we add such a prefix to the arguments of the `ipmitool` command:

`-I lanplus -U USER -P PASSWORD -H HOST`

**Comment by duecedriver** [comment]
the longterm goal should be to decouple the sensor loop from the hardware/container its being run on..

this way not only on localhost can it be more effective, but it can monitor servers out-of-band from another machine

sensor data can be gathered many ways through remote commands

ipmi calls 
use plink and you can run ssh commands on remote systems 

etc

**Comment by petersulyok** [comment]
Hi @davidnghongkong, @bockbilbo, @alexanderek 

I've added a new parameter to the config file where you can specify remote access parameters e.g.:

`-U user -P password -H host`

Could please checkout the current state of the github repository and test it? Practically you need only `smfc.py` and the new parameter (`[Ipmi] remote_parameters=`) in the config file.

**Comment by petersulyok** [comment]
Implemented in v3.8.0
